operator/apis/http/v1alpha1/httpscalingset_types.go

```diff
@@ -0,0 +1,234 @@
+	// queryable and should be preserved when modifying objects.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// +kubebuilder:default=default
+	// +optional
+	// Name of the service account to be used
+	ServiceAccountName string `json:"serviceAccountName"`
+}
+
+// HTTPScalerConfigurationSepc defines the desired state of scaler configuration
+type HTTPScalerConfigurationSepc struct {
+	// +kubebuilder:default=9090
+	// +optional
+	// Port to be used for proxy operations
+	Port *int32 `json:"port,omitempty"`
+}
+
+// HTTPScalerSepc defines the desired state of Scaler component
+type HTTPScalerSepc struct {
+	// +kubebuilder:default={}
+	// +optional
+	// Traffic configuration
+	Config HTTPScalerConfigurationSepc `json:"config,omitempty"`
+	// Number of replicas for the interceptor
+	Replicas *int32 `json:"replicas,omitempty"`
+	// Container image name.
+	// +optional
+	Image *string `json:"image,omitempty"`
+	// ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.
+	// More info: https://kubernetes.io/docs/concepts/containers/images#specifying-imagepullsecrets-on-a-pod
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	// Compute Resources required by this scaler.
+	// Cannot be updated.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+	// Map of string keys and values that can be used to organize and categorize
+	// (scope and select) objects. May match selectors of replication controllers
+	// and services.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+	// Annotations is an unstructured key value map stored with a resource that may be
+	// set by external tools to store and retrieve arbitrary metadata. They are not
+	// queryable and should be preserved when modifying objects.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+	// +kubebuilder:default=default
+	// +optional
+	// Name of the service account to be used
+	ServiceAccountName string `json:"serviceAccountName"`
+}
+
+// HTTPScalingSetSpec defines the desired state of HTTPScalingSet
+type HTTPScalingSetSpec struct {
+	Interceptor HTTPInterceptorSepc `json:"interceptor"`
+	Scaler      HTTPScalerSepc      `json:"scaler"`
+}
+
+// HTTPScalingSetStatus defines the observed state of HTTPScalingSet
+type HTTPScalingSetStatus struct{}
```

Comment:
I am sure this is empty only because the PR is still WIP but for discussion sake mentioning controller may want to capture some conditions here, such as:
* unable to create interceptor/scaler
  * kube-api rejects some resource creation
  * failure to form the `externalScalerURI`
* interceptor/scaler are not running with desired replica count and/or are unhealthy
* possibly getting some basic stats from interceptor admin endpoint?